File: JavaFX/src/main/java/Entities/Livraison.java
package Entities;

import java.time.LocalDate;

public class Livraison {
    private int idLivraison;
    private int idCommande; // Clé étrangère pour la jointure 1-to-1
    private String numeroBL; // Bon de livraison
    private LocalDate dateLivraison;
    private String livreur;
    private String statutLivraison; // En cours, Livré, Retardé
    private String noteDelivery;
    private double latitude; // Nouvelle variable pour la latitude
    private double longitude; // Nouvelle variable pour la longitude

    // Constructeurs
    public Livraison() {
    }

    public Livraison(int idCommande, String numeroBL, LocalDate dateLivraison,
                     String livreur, String statutLivraison, String noteDelivery) {
        this.idCommande = idCommande;
        this.numeroBL = numeroBL;
        this.dateLivraison = dateLivraison;
        this.livreur = livreur;
        this.statutLivraison = statutLivraison;
        this.noteDelivery = noteDelivery;
    }

    public Livraison(int idCommande, String numeroBL, LocalDate dateLivraison,
                     String livreur, String statutLivraison, String noteDelivery,
                     double latitude, double longitude) {
        this.idCommande = idCommande;
        this.numeroBL = numeroBL;
        this.dateLivraison = dateLivraison;
        this.livreur = livreur;
        this.statutLivraison = statutLivraison;
        this.noteDelivery = noteDelivery;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public Livraison(int idLivraison, int idCommande, String numeroBL, LocalDate dateLivraison,
                     String livreur, String statutLivraison, String noteDelivery,
                     double latitude, double longitude) {
        this.idLivraison = idLivraison;
        this.idCommande = idCommande;
        this.numeroBL = numeroBL;
        this.dateLivraison = dateLivraison;
        this.livreur = livreur;
        this.statutLivraison = statutLivraison;
        this.noteDelivery = noteDelivery;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    // Getters et Setters
    public int getIdLivraison() {
        return idLivraison;
    }

    public void setIdLivraison(int idLivraison) {
        this.idLivraison = idLivraison;
    }

    public int getIdCommande() {
        return idCommande;
    }

    public void setIdCommande(int idCommande) {
        this.idCommande = idCommande;
    }

    public String getNumeroBL() {
        return numeroBL;
    }

    public void setNumeroBL(String numeroBL) {
        this.numeroBL = numeroBL;
    }

    public LocalDate getDateLivraison() {
        return dateLivraison;
    }

    public void setDateLivraison(LocalDate dateLivraison) {
        this.dateLivraison = dateLivraison;
    }

    public String getLivreur() {
        return livreur;
    }

    public void setLivreur(String livreur) {
        this.livreur = livreur;
    }

    public String getStatutLivraison() {
        return statutLivraison;
    }

    public void setStatutLivraison(String statutLivraison) {
        this.statutLivraison = statutLivraison;
    }

    public String getNoteDelivery() {
        return noteDelivery;
    }

    public void setNoteDelivery(String noteDelivery) {
        this.noteDelivery = noteDelivery;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public boolean hasLocation() {
        return latitude != 0 && longitude != 0;
    }

    @Override
    public String toString() {
        return "Livraison{" +
                "idLivraison=" + idLivraison +
                ", idCommande=" + idCommande +
                ", numeroBL='" + numeroBL + '\'' +
                ", dateLivraison=" + dateLivraison +
                ", livreur='" + livreur + '\'' +
                ", statutLivraison='" + statutLivraison + '\'' +
                ", noteDelivery='" + noteDelivery + '\'' +
                ", latitude=" + latitude +
                ", longitude=" + longitude +
                '}';
    }
}

File: web/src/Entity/Livraison.php
<?php

namespace App\Entity;

use App\Repository\LivraisonRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: LivraisonRepository::class)]
#[ORM\Table(name: 'livraison')]
class Livraison
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'idLivraison')]
    private ?int $idLivraison = null;

    #[ORM\OneToOne(inversedBy: 'livraison')]
    #[ORM\JoinColumn(name: 'idCommande', referencedColumnName: 'idCommande')]
    private ?Commande $commande = null;

    #[ORM\Column(name: 'numeroBL', length: 50)]
    private ?string $numeroBL = null;

    #[ORM\Column(name: 'dateLivraison', type: 'date')]
    private ?\DateTimeInterface $dateLivraison = null;

    #[ORM\Column(length: 100)]
    private ?string $livreur = null;

    #[ORM\Column(name: 'statutLivraison', length: 50)]
    private ?string $statutLivraison = null;

    #[ORM\Column(name: 'noteDelivery', length: 255, nullable: true)]
    private ?string $noteDelivery = null;

    #[ORM\Column(type: 'float')]
    private ?float $latitude = 0.0;

    #[ORM\Column(type: 'float')]
    private ?float $longitude = 0.0;

    public function getIdLivraison(): ?int
    {
        return $this->idLivraison;
    }

    public function getCommande(): ?Commande
    {
        return $this->commande;
    }

    public function setCommande(?Commande $commande): static
    {
        $this->commande = $commande;
        return $this;
    }

    public function getNumeroBL(): ?string
    {
        return $this->numeroBL;
    }

    public function setNumeroBL(string $numeroBL): static
    {
        $this->numeroBL = $numeroBL;
        return $this;
    }

    public function getDateLivraison(): ?\DateTimeInterface
    {
        return $this->dateLivraison;
    }

    public function setDateLivraison(\DateTimeInterface $dateLivraison): static
    {
        $this->dateLivraison = $dateLivraison;
        return $this;
    }

    public function getLivreur(): ?string
    {
        return $this->livreur;
    }

    public function setLivreur(string $livreur): static
    {
        $this->livreur = $livreur;
        return $this;
    }

    public function getStatutLivraison(): ?string
    {
        return $this->statutLivraison;
    }

    public function setStatutLivraison(string $statutLivraison): static
    {
        $this->statutLivraison = $statutLivraison;
        return $this;
    }

    public function getNoteDelivery(): ?string
    {
        return $this->noteDelivery;
    }

    public function setNoteDelivery(?string $noteDelivery): static
    {
        $this->noteDelivery = $noteDelivery;
        return $this;
    }

    public function getLatitude(): ?float
    {
        return $this->latitude;
    }

    public function setLatitude(float $latitude): static
    {
        $this->latitude = $latitude;
        return $this;
    }

    public function getLongitude(): ?float
    {
        return $this->longitude;
    }

    public function setLongitude(float $longitude): static
    {
        $this->longitude = $longitude;
        return $this;
    }

    public function hasLocation(): bool
    {
        return $this->latitude != 0 && $this->longitude != 0;
    }
}
